spec/javascripts/actions/projectBoard_spec.js

```diff
@@ -1,30 +1,92 @@
-import { expandStoryIfNeeded } from '../../../app/assets/javascripts/actions/projectBoard';
-import { toggleStory } from '../../../app/assets/javascripts/actions/story';
+import { 
+  expandStoryIfNeeded, closeSearch, 
+  closeSearchSuccess, sendErrorNotificationIfNeeded
+} from '../../../app/assets/javascripts/actions/projectBoard';
+import { toggleStory, receiveStories } from '../../../app/assets/javascripts/actions/story';
+import { addNotification } from '../../../app/assets/javascripts/actions/notifications';
 
 describe('Project Board Actions', () => {
-  it('Should dispatch toggleStory when storyId is true', async () => {
-    const storyId = 127;
+  describe('expandStoryIfNeeded', () => {
+    describe('when storyId is true', () => {
+      it('Should dispatch toggleStory', () => {
+        const storyId = 127;
+    
+        const fakeGetHash = sinon.stub();
+        fakeGetHash.returns(storyId);
+    
+        const fakeDispatch = sinon.stub();
+        fakeDispatch.resolves({});
+    
+        expandStoryIfNeeded(fakeDispatch, fakeGetHash);
+        expect(fakeDispatch).toHaveBeenCalledWith(toggleStory(storyId));
+      });
+    });
+  
+    describe('when storyId is false', () => {
+      it('Should not dispatch toggleStory', () => {
+        const storyId = null;
+    
+        const fakeGetHash = sinon.stub();
+        fakeGetHash.returns(storyId);
+    
+        const fakeDispatch = sinon.stub();
+        fakeDispatch.resolves({});
+    
+        expandStoryIfNeeded(fakeDispatch, fakeGetHash);
+        expect(fakeDispatch).not.toHaveBeenCalledWith(toggleStory(storyId));
+      });
+    });
+  });
 
-    const fakeGetHash = sinon.stub();
-    fakeGetHash.returns(storyId);
+  describe('sendErrorNotificationIfNeeded', () => {
+    describe('when condition is false', () => {
+      it('Calls "dispatch"', () => {
+        const condition = false;
+        const code = 'code';
+  
+        const fakeDispatch = sinon.stub();
+        fakeDispatch.resolves({});
+  
+        sendErrorNotificationIfNeeded(fakeDispatch, code, condition);
+        expect(fakeDispatch).toHaveBeenCalled();
```

Comment:
There are some arguments when with trigger dispatch? Ex.:
```js
dispatch('some-value')
```
If so, you could improve this test with `toHaveBeenCalledWith`:
```js
expect(fakeDispatch).toHaveBeenCalledWith('some-value');
```

Reply:
Yes, but the dispatch the function than will be called, it's call other function, so I could not made all injections of dependencies.